connectivity/FEATURE_BLE/include/ble/SecurityManager.h

```diff
@@ -501,11 +501,29 @@ class SecurityManager
      * Normally all bonding information is lost when device is reset, this requests that the stack
      * attempts to save the information and reload it during initialisation. This is not guaranteed.
      *
+     * @note This option is itself saved together with bonding data. When data is read after reset,
+     * the state of this option decides if data should be restored. If this option has not been saved
```

Comment:
My read of this is that `writeBondingStateToPersistentStorage` is effectively a no-op unless this is called before the reset. What's the rationale behind this?

Reply:
preserveBondingStateOnReset is an option that affects behaviour after the reset. Maybe I should've called it loadBondingDataFromPersistentStorageAfterReset

Reply:
Could `writeBondingStateToPersistentStorage` implicitly set `preserveBondingStateOnReset`? Or is there a reason someone would write the data but not automatically load it after a reset?

Reply:
That is a good point. If I can't find a good reason by tomorrow I'll add the call.